scg-system/src/main/java/io/telicent/backup/services/DatasetBackupService.java

```diff
@@ -756,4 +759,40 @@ private void compressAndStoreBackupMetadata(ObjectNode response, String dirPath)
         writeObjectNodeToFile(response, dirPath + JSON_INFO_SUFFIX);
     }
 
+    /**
+     * A utility method that takes the size of something in bytes and represents it in
+     * a more easy to grasp format.
+     * @param bytes size of file in question
+     * @return a string representation in KB, MB, GB, TB or PB
+     */
+    public static String humanReadableByteCount(long bytes) {
+        int unit = 1000;
+        if (bytes < unit) return bytes + "B";
+        int exp = (int) (Math.log(bytes) / Math.log(unit));
+        String pre = "kMGTP".charAt(exp - 1) + "";
+        return String.format("%.1f%sB", bytes / Math.pow(unit, exp), pre);
+    }
+
+    /**
+     * A utility method that takes the duration and represents it in
+     * a more easy to grasp format.
+     * @param duration the number of milliseconds in question
+     * @return a string representation in hours. minutes, seconds and milliseconds
+     */
+    public static String humanReadableDuration(Duration duration) {
+        long hours = duration.toHours();
+        long minutes = duration.toMinutes() % 60;
+        long seconds = duration.getSeconds() % 60;
```

Comment:
```suggestion
        long seconds = duration.toSeconds() % 60;
```

For consistency/correctness?

Reply:
Agreed.